build-n-publish.sh

```diff
@@ -0,0 +1,120 @@
+  exit 1
+fi
+
+# Stop script on NZEC
+set -e
+# Stop script if unbound variable found (use ${var:-} if intentional)
+set -u
+
+# Lines added to get the script running in the script path shell context
+# reference: http://www.ostricher.com/2014/10/the-right-way-to-get-the-directory-of-a-bash-script/
+cd $(dirname $0)
+
+# To avoid issues with MINGW and Git Bash, see:
+# https://github.com/docker/toolbox/issues/673
+# https://gist.github.com/borekb/cb1536a3685ca6fc0ad9a028e6a959e3
+export MSYS_NO_PATHCONV=1
+export MSYS2_ARG_CONV_EXCL="*"
+
+versionBuild=${commit}
+# Ugly code to deal with versions
+# Input:
+#   version=v12.34.5
+#   versionBuild=94f85efb9c
+#   versionPre=0pr
+# Output:
+#   preReleasePrefix= "pre-" | ""
+#   versionMayor=v12
+#   versionMayorMinor=v12.34
+#   versionMayorMinorPatch=v12.34.5
+#   versionMayorMinorPatchPre=v12.34.5-0pr
+#   versionFull=v12.34.5-0pr+94f85efb9c
+#   versionFullForTag=v12.34.5-0pr_94f85efb9c
+# region Ugly code to deal with versions
+
+versionFull=${version}
+
+if [ ! -z ${versionPre} ]
+then
+  versionFull=${versionFull}-${versionPre}
+fi
+
+if [ ! -z ${versionBuild} ]
+then
+  versionFull=${versionFull}+${versionBuild}
+fi
+
+# https://semver.org/spec/v2.0.0.html#backusnaur-form-grammar-for-valid-semver-versions
+# <valid semver> ::= <version core>
+#                  | <version core> "-" <pre-release>
+#                  | <version core> "+" <build>
+#                  | <version core> "-" <pre-release> "+" <build>
+#
+# <version core> ::= <major> "." <minor> "." <patch>
+versionBuild="$(echo ${versionFull}+ | cut -d'+' -f2)"
+versionMayorMinorPatchPre="$(echo ${versionFull} | cut -d'+' -f1)" # v0.0.0-xxx (ignoring `+` if exists)
+versionPre="$(echo ${versionMayorMinorPatchPre}- | cut -d'-' -f2)"
+versionMayorMinorPatch="$(echo ${versionMayorMinorPatchPre} | cut -d'-' -f1)" # v0.0.0 (ignoring `-` if exists)
+versionMayor="$(echo ${versionMayorMinorPatch} | cut -d'.' -f1)" # v0
+versionMinor="$(echo ${versionMayorMinorPatch} | cut -d'.' -f2)"
+versionMayorMinor=${versionMayor}.${versionMinor} # v0.0
+versionPatch="$(echo ${versionMayorMinorPatch} | cut -d'.' -f3)"
+
+if [ -z ${versionBuild} ]
+then
+  versionFullForTag=${versionMayorMinorPatchPre}
+else
+  versionFullForTag=${versionMayorMinorPatchPre}_${versionBuild}
+fi
+
+if [ ! -z ${versionPre} ]
+then
+  preReleasePrefix="pre-"
+else
+  preReleasePrefix=""
+fi
+# endregion Ugly code to deal with versions
+
+imageName=fromdoppler/doppler-currency
+canonicalTag=${preReleasePrefix}${versionFullForTag}
+
+docker build \
+    -t ${imageName}:${canonicalTag} \
+    --build-arg EncryptionSettings__InitVectorAsAsciiString="$EncryptionSettings__InitVectorAsAsciiString" \
+    --build-arg EncryptionSettings__SaltValueAsAsciiString="$EncryptionSettings__SaltValueAsAsciiString" \
+    --build-arg EncryptionSettings__Password="$EncryptionSettings__Password" \
+    --build-arg version=${preReleasePrefix}${versionFull} \
```

Comment:
I think that these lines are not required

Reply:
O, the last one is necessary for this:

https://github.com/MakingSense/doppler-forms/blob/master/Dockerfile#L39-L41

So, keep it in this way:

```
docker build \
    -t ${imageName}:${canonicalTag} \
    --build-arg version=${preReleasePrefix}${versionFull} \
    .
```

Reply:
Ok, I'll add line 39 and 40 in Dockerfile file and remove unnecessary lines 102, 103 and 104 in build-n-publish.sh